sql_metadb/derived_tables/feesfines_accounts_actions.sql

```diff
@@ -1,44 +1,96 @@
     jsonb_extract_path_text(fa.jsonb, 'userId')::uuid AS account_user_id,
     ff.id AS transaction_id,
-    jsonb_extract_path_text(ff.jsonb, 'accountId')::uuid AS account_id,
     jsonb_extract_path_text(ff.jsonb, 'amountAction')::numeric(12,2) AS transaction_amount,
     jsonb_extract_path_text(ff.jsonb, 'balance')::numeric(12,2) AS account_balance,
     jsonb_extract_path_text(ff.jsonb, 'typeAction') AS type_action,
     jsonb_extract_path_text(ff.jsonb, 'dateAction')::timestamptz AS transaction_date,
     jsonb_extract_path_text(ff.jsonb, 'createdAt') AS transaction_location,
     jsonb_extract_path_text(ff.jsonb, 'transactionInformation') AS transaction_information,
-    jsonb_extract_path_text(ff.jsonb, 'source') AS operator_id,
+    jsonb_extract_path_text(ff.jsonb, 'source') AS action_created_by,
     jsonb_extract_path_text(ff.jsonb, 'paymentMethod') AS payment_method,
     uu.id AS user_id,
     uu.patron_group AS user_patron_group_id,
-    ug.id AS patron_group_id,
-    ug.group AS patron_group_name
+    ug.group AS patron_group_name,
+CASE WHEN
+    jsonb_extract_path_text(ff.jsonb, 'typeAction') IN
+    ('Paid partially','Paid fully','Waived partially','Waived fully','Credited partially','Credited fully')
+    THEN jsonb_extract_path_text(ff.jsonb, 'amountAction')::numeric(12,2) * -1
+    ELSE jsonb_extract_path_text(ff.jsonb, 'amountAction')::numeric(12,2)
+    END AS signed_transaction_amount
 FROM
     folio_feesfines.accounts AS fa
     LEFT JOIN folio_feesfines.feefineactions AS ff ON fa.id = jsonb_extract_path_text(ff.jsonb, 'accountId')::uuid
     LEFT JOIN folio_users.users__t AS uu ON jsonb_extract_path_text(fa.jsonb, 'userId')::uuid = uu.id
     LEFT JOIN folio_users.groups__t AS ug ON uu.patron_group = ug.id
 ORDER BY fine_account_id, transaction_date;
 
+COMMENT ON COLUMN feesfines_accounts_actions.fine_account_id IS 'User fine/fee account id, UUID'
+
+COMMENT ON COLUMN feesfines_accounts_actions.fine_account_amount IS 'Amount of the fine/fee'
+
+COMMENT ON COLUMN feesfines_accounts_actions.fine_date IS 'Date and time the account of the fine/fee was created'
+
+COMMENT ON COLUMN feesfines_accounts_actions.fine_updated_date IS 'Date and time the account of the fine/fee was updated'
+
+COMMENT ON COLUMN feesfines_accounts_actions.fee_fine_id IS 'ID of the fee/fine'
+
+COMMENT ON COLUMN feesfines_accounts_actions.owner_id IS 'ID of the account owner'
+
+COMMENT ON COLUMN feesfines_accounts_actions.fee_fine_owner IS 'Owner of the account'
+
+COMMENT ON COLUMN feesfines_accounts_actions.fee_fine_type IS 'Fee/fine that is up to the desecration of the user'
+
+COMMENT ON COLUMN feesfines_accounts_actions.material_type_id IS 'ID of the material type of the item'
+
+COMMENT ON COLUMN feesfines_accounts_actions.material_type IS 'Material type of the item'
+
+COMMENT ON COLUMN feesfines_accounts_actions.payment_status IS 'Overall status of the payment/waive/transfer/refund/cancel'
```

Comment:
wrong column name. Did you mean "payment_status_name"?

Reply:
changed the name of the column in main query to just "payment_status"